fastatophy/fastatophy.go

```diff
@@ -0,0 +1,99 @@
+// Copyright ©2017 The bíogo Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// fastatophy converts a multiple-sequence alignment in
+// FASTA to PHYLIP (sequential) format.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/biogo/biogo/alphabet"
+	"github.com/biogo/biogo/io/seqio"
+	"github.com/biogo/biogo/io/seqio/fasta"
+	"github.com/biogo/biogo/seq/linear"
+)
+
+var (
+	inf  = flag.String("in", "", "input FASTA filename")
+	outf = flag.String("out", "", "output PHYLIP filename")
+	help = flag.Bool("help", false, "help prints this message")
+)
+
+func main() {
+	var (
+		n, alnLen  int
+		strictName string
+	)
+
+	flag.Parse()
+	if *help {
+		flag.Usage()
+		os.Exit(0)
+	}
+
+	in, err := os.Open(*inf)
+	if err != nil {
+		log.Fatalf("failed to open FASTA file: %v", err)
+	}
+	defer in.Close()
+	r := fasta.NewReader(in, linear.NewSeq("", nil, alphabet.Protein))
+
+	out, err := os.Create(*outf)
+	if err != nil {
+		log.Fatalf("failed to open PHYLIP file: %v", err)
+	}
+	defer out.Close()
+
+	// Read all FASTA records to get total number of sequences
+	// (n) and length of each sequence. alnLen stores the
+	// sequence length of the last record.
+	n = 0
+	sc := seqio.NewScanner(r)
+	for sc.Next() {
+		s := sc.Seq()
+		alnLen = s.Len()
```

Comment:
This still needs a check. Just assigning to a var in a loop is a bad code smell - I would check that they agree.